### PR TITLE
[MCC-139310] Feature/moya

### PR DESCRIPTION
**Pull request by csavage-mdsol**
Sorry about the mega PR.  I haven't really thought of how swap an app in bite size chunks.  This gets the previous specs working with a new rails app, so it basically implements a simplified index, a show, and a root, as well as the magic for external initializers.  I changed a few things about the drd model.  I changed `uuid` to `id` because I always thought "I don't care that its a uuid, its an id that happens to be a uuid".  To this end, I also included ActiveUUID which seemed like a moderately good idea.  I also changed `built_at` to `created_at` so we could just use timestamps like normal people.  

A few weird things popped up during this:
1) Crichton has a terrible error if your api descriptors aren't where it is expecting them.  it raises an `undefined method transitions on nil`
2) The ActiveUUID thing started requiring an actual UUID object at some point.  Not sure what I broke.  It should take a string.  Left a todo for a fix.
3) On the index route, I noticed that if you hand crichton a `{ state: :collection}` option it would pass that option on to the drds in `items`, which would explode, because there isn't a state `collection`.  Probably related to the cascading permissions issue.
4) If you hand an ActiveRecord::Relation in instead of an array of instantiated objects, things go boom.

Opened against develop, but I created a branch `crichton_test_service` that we can keep using until Moya is ready to starburst.

Comments and criticism most welcome.  I'd like us to like this.
@yzhangmedidata @HonoreDB @sheavalentine-mdsol 

```
Finished in 33.14 seconds (files took 0.99444 seconds to load)
12 examples, 0 failures, 6 pending
```

### CODE REVIEW

File: lib/moya/app/controllers/application_controller.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class ApplicationController < ActionController::Base
+  # Prevent CSRF attacks by raising an exception.
+  # For APIs, you may want to use :null_session instead.
+  protect_from_forgery with: :exception
```
Comment (HonoreDB) [inline lib/moya/app/controllers/application_controller.rb]: For APIs, apparently, we may want to use :null_session instead.

File: lib/moya/app/controllers/drds_controller.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  ]
+
+  def index
+    @drds = Drd.all
+
+    filtering_params(params).each do |k,v|
+      @drds = @drds.public_send(key, value) if value.present?
+    end
+
+    # TODO: Should this be fixed in Crichton?
```
Comment (HonoreDB) [inline lib/moya/app/controllers/drds_controller.rb]: Agree, this should be fixed in Crichton.

**Comment by HonoreDB** [comment]
Looks great to me.

File: lib/moya/app/controllers/application_controller.rb (hunk 1 above)
Comment (csavage-mdsol) [inline lib/moya/app/controllers/application_controller.rb]: Do you have a good source for this so I could include it with a comment?

Comment (csavage-mdsol) [inline lib/moya/app/controllers/application_controller.rb]: Im dumb.

Comment (csavage-mdsol) [inline lib/moya/app/controllers/application_controller.rb]: done

File: .gitignore [hunk 3]
```diff
@@ -22,4 +22,5 @@ tmp
 *.sqlite3
 development.log
 mkmf.log
-lib/crichton_test_service/config/crichton.yml
+lib/moya/config/crichton.yml
```
Comment (sheavalentine-mdsol) [inline .gitignore:25]: Why is this ignored?

Comment (csavage-mdsol) [inline .gitignore:25]: It is regenerated every time a rails process is spun up, and is dependent on the specified port.  If we checked it in, we'd be stuck with a single port.

Comment (miwest929) [inline .gitignore:25]: dicebag

**Comment by sheavalentine-mdsol** [comment]
Okay, LGTM, merging.

**Pull request merged**